Slider/Assets/Scripts/Player/Player.cs

```diff
@@ -133,6 +136,27 @@ private void FixedUpdate()
             transform.SetParent(null);
         }
     }
+    //Jroo: Either says "Keyboard Mouse" or "Controller" based on last input
+    public string GetCurrentControlScheme()
+    {
+        return playerInput.currentControlScheme;
+    }
+    public event Action<string> ControlSchemeChanged;
```

Comment:
in general in slider codebase, put events at the top and prefix it with "On[WhateverHappens]" 
also to match the rest of code it should prob use an EventHandler instead of Action
so something like 
```cs
public static event System.EventHandler<System.EventArgs> OnControlSchemeChanged;
```

in order to pass a string you'll need to make a new class that extends EventArgs too, i.e.:
```cs
public class ControlsChangedEventArgs : System.EventArgs 
{
   string newScheme;
}
```
and when calling it you'll do
```cs
OnControlSchemeChanged?.Invoke(this, new ControlsChangedEventArgs { newScheme = newControlScheme} );
```